c/shared/source/tx_shared/tx_shared.c

```diff
@@ -2155,10 +2155,14 @@ static int t1_GlyphBeg(abfGlyphCallbacks *cb, abfGlyphInfo *info) {
             fatal(h, "selected glyphs span multiple FDs");
     }
     /* Create glyph name */
-    if (info->cid == 0)
-        strcpy(gname, ".notdef");
-    else
-        sprintf(gname, "cid%hu", info->cid);
+    if (info->gname.ptr != NULL) {
+        strcpy(gname, info->gname.ptr);
```

Comment:
Using strncpy seems particularly wise here given the fixed buffer size and the potentially unverified length.

Reply:
Updated.